test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ConformanceComparison.kt

```diff
@@ -4,98 +4,223 @@ import com.amazon.ionelement.api.loadSingleElement
+    ReportAnalyzer(first, second) {
+    override val reportTitle: String = "Conformance comparison report-Cross Commit-${first.engine.uppercase()}"
+    override fun generateComparisonReport(limit: Int) =
+        buildString {
+            this.appendLine(
+                """### $reportTitle
+| | Base (${first.commitId}) | ${second.commitId} | +/- |
 | --- | ---: | ---: | ---: |
-| % Passing | ${"%.2f".format(origPassingPercent)}% | ${"%.2f".format(newPassingPercent)}% | ${"%.2f".format(newPassingPercent - origPassingPercent)}% |
-| :white_check_mark: Passing | $numOrigPassing | $numNewPassing | ${numNewPassing - numOrigPassing} |
-| :x: Failing | $numOrigFailing | $numNewFailing | ${numNewFailing - numOrigFailing} |
-| :large_orange_diamond: Ignored | $numOrigIgnored | $numNewIgnored | ${numNewIgnored - numOrigIgnored} |
-| Total Tests | $totalOrig | $totalNew | ${totalNew - totalOrig} |
-""",
-    )
-
-    comparisonReportFile.appendText(
-"""
+| % Passing | ${"%.2f".format(firstPassingPercent)}% | ${"%.2f".format(secondPassingPercent)}% | ${"%.2f".format(secondPassingPercent - firstPassingPercent)}% |
+| :white_check_mark: Passing | $firstPassingSize | $secondPassingSize | ${secondPassingSize - firstPassingSize} |
+| :x: Failing | $firstFailingSize | $secondFailingSize | ${secondFailingSize - firstFailingSize} |
+| :large_orange_diamond: Ignored | $firstIgnoreSize | $secondIgnoreSize | ${secondIgnoreSize - firstIgnoreSize} |
+| Total Tests | $firstTotalSize | $secondTotalSize | ${secondTotalSize - firstTotalSize} |
+                """.trimIndent()
+            )
+            this.appendLine(
+                """
 Number passing in both: ${passingInBoth.count()}
 
 Number failing in both: ${failingInBoth.count()}
 
-Number passing in Base ($origCommitId) but now fail: ${passingOrigFailingNew.count()}
+Number passing in Base (${first.commitId}) but now fail: ${passingFirstFailingSecond.count()}
+
+Number failing in Base (${first.commitId}) but now pass: ${failureFirstPassingSecond.count()}
+                """.trimIndent()
+            )
+            if (passingFirstFailingSecond.isNotEmpty()) {
+                // character count limitation with comments in GitHub
+                // also, not ideal to list out hundreds of test names
+                if (passingFirstFailingSecond.size < limit) {
+                    this.appendLine(":interrobang: CONFORMANCE REPORT REGRESSION DETECTED :interrobang:. The following test(s) were previously passing but now fail:\n<details><summary>Click here to see</summary>\n\n")
+
+                    passingFirstFailingSecond.forEach { testName ->
+                        this.appendLine("- $testName")
+                    }
+                    this.appendLine("</details>")
+                } else {
+                    this.appendLine(":interrobang: CONFORMANCE REPORT REGRESSION DETECTED :interrobang:")
+                    this.appendLine("Download Artifact from Summary to view the complete list")
```

Comment:
Can also look at the CI's "event summary" rather than needing to download the artifact. E.g. https://github.com/yliuuuu/partiql-lang-kotlin/actions/runs/7483462300/attempts/1#summary-20368831620 shows all the failing tests in eval that pass in legacy.